### ISSUE TITLE
Why doesn't `asyncio.as_completed` cancel unfinished tasks if it timeout?

### ISSUE THREAD

**Issue by NewUserHa** [python/cpython#99933]
# Feature or enhancement
`asyncio.as_completed` should cancel unfinished tasks if it timeout, or even maybe have the ability to cancel no needed task

# Pitch
https://github.com/python/cpython/blob/0563be23a557917228a8b48cbb31bda285a3a815/Lib/asyncio/tasks.py#L588-L609
Why doesn't `asyncio.as_completed` cancel unfinished tasks if it timeout?
Maybe say there may be many network requests pending, since that already raised exception then future requests should probably be canceled to save resources.
Probably because the `concurrent.futures.as_complete` didn't cancel unfinished tasks? But it's local resources than asyncio is usually network resources.

More, I know the `asyncio.wait()` can do cancel the pending job. But `asyncio.as_complete()` probably should also be able to cancel the next coros if got the wanted result. But currently, the returned is a coroutine that can't cancel.

**Comment by gvanrossum** [comment]
This is indeed unfortunate, but we cannot change it without breaking backwards compatibility. I recommend trying to work around it using task groups.

**Comment by NewUserHa** [comment]
But that can definitely waste unnecessary network resources if users input many awaitables to the `asyncio.as_completed()` isn't it?

Should the document mention that situation for the users who never read the source code? Or recommend using 'task group' in the document of `asyncio.as_completed()` as well? (currently, only `asyncio.gather()` recommend using 'task group')

**Comment by gvanrossum** [comment]
I agree that we could add a note to the docs for `as_completed`. If we wanted to recommend task groups we'd at least have to give a code sample showing how to do that -- it's not completely trivial.
